convert/convert.go

```diff
@@ -61,6 +75,52 @@ func Convert(inputFilename, outputFilename string, from, to Format) error {
 	return nil
 }
 
+func convertKongGateway2xTo3x(input *file.Content) (*file.Content, error) {
+	if input == nil {
+		return nil, fmt.Errorf("input content is nil")
+	}
+	outputContent := input.DeepCopy()
+
+	convertedRoutes := []*file.FRoute{}
+	for _, service := range outputContent.Services {
+		for _, route := range service.Routes {
+			convertedRoutes = append(convertedRoutes, migrateRoutesPathFieldPre300(route))
+		}
+		service.Routes = convertedRoutes
+	}
+
+	if outputContent.FormatVersion != "3.0" {
```

Comment:
This message is skipped if the user includes `_format_version: "3.0"` in their input file. It is incorrect input but still possible. 

Input
```
_format_version: "3.0"
services:
- name: svc1
  host: mockbin.org
  path: /status/200
  routes:
  - name: r1
    paths:
    - /status/\d+
    - /code/\d+
```

Output
```
_format_version: "3.0"
services:
- host: mockbin.org
  name: svc1
  path: /status/200
  routes:
  - name: r1
    paths:
    - ~/status/\d+
    - ~/code/\d+
```

Reply:
This is also possible: `From the config file, the _format_version field has been migrated from 'bacon' to '3.0'.`

Reply:
Actually nevermind on that second point. [I see validation occurs](https://github.com/Kong/deck/blob/main/cmd/convert.go#L29), just not in these tests.

Reply:
What behavior do we want if the user provides an input format command line arg that is different than the `_format_version` contained in the file(s)? Is it possible that a user will have multiple formats in use across files and want to upgrade them all to 3.x? I'm going off of [this comment](https://github.com/Kong/deck/pull/753/files#r971338349) for the multi-file use case.

Reply:
This warning is printed when a `_format_version` is migrated from something different than `3.0` to `3.0`, so why should decK print it anyway even if the `_format_version` already had the correct state?

In case of multiple files, this would look like this:

```
$ cat kong.yaml
_format_version: "3.0"
services:
- host: mockbin.org
  name: svc1
  path: /status/200
  routes:
  - name: r1
    paths:
    - /status/\d+
    - /code/\d+

$ cat kong1.yaml
_format_version: "1.0"
services:
- host: mockbin.org
  name: svc2
  path: /status/200
  routes:
  - name: r2
    paths:
    - /status/\d+
```

```
$ /tmp/deck convert --from kong-gateway-2.x --to kong-gateway-3.x

From the '/tmp/statefiles/kong.yaml' config file, a regex pattern usage was detected
for the '/status/\d+, /code/\d+' paths in the 'paths' field.
Kong gateway versions 3.0 and above require that regular expressions
start with a '~' character to distinguish from simple prefix match.
In order to make these paths compatible with 3.x, a '~' prefix has been added.

From the '/tmp/statefiles/kong1.yaml' config file, a regex pattern usage was detected
for the '/status/\d+' paths in the 'paths' field.
Kong gateway versions 3.0 and above require that regular expressions
start with a '~' character to distinguish from simple prefix match.
In order to make these paths compatible with 3.x, a '~' prefix has been added.

From the '/tmp/statefiles/kong1.yaml' config file, the _format_version field has been migrated from '1.0' to '3.0'.
These automatic changes may not be fully correct or exhaustive, so please
take the time to perform a manual audit of the config file.

For related information, please visit: https://docs.konghq.com/deck/latest/3.0-upgrade
```